pkg/registry/minion/rest.go

```diff
@@ -143,15 +143,15 @@ func ResourceLocation(getter ResourceGetter, connection client.ConnectionInfoGet
 	node := nodeObj.(*api.Node)
 	host := node.Name // TODO: use node's IP, don't expect the name to resolve.
 
-	if portReq != "" {
```

Comment:
I think the idea here was to only use the kubelet transport if we're going to talk to kubelet; presumably if they set a port, they want to talk to something other than kubelet and then we have no idea what transport would be appropriate.